index.js

```diff
@@ -762,19 +752,7 @@ function parse (args, opts) {
     return isSet
   }
 
-  function setDefaulted (key) {
```

Comment:
I'm concerned that we're completely removing the `setDefaulted` functionality, which I see used here:

```js
        if (((configOnly && flags.configs[keys.join('.')]) || !configOnly) && (!hasKey(argv, keys) || flags.defaulted[keys.join('.')])) {
          setArg(keys.join('.'), process.env[envVar])
        }
```

and, here:

```js
        if (!hasKey(argv, fullKey.split('.')) || (flags.defaulted[fullKey]) || (flags.arrays[fullKey] && configuration['combine-arrays'])) {
          setArg(fullKey, value)
        }
```

If it's true that we're not using this functionality, awesome let's get rid of it -- I'm just slightly worried we might be missing a unit tests for this historic behavior ... I tried to perform a `git blame` and it looks like this feature has been around forever.

Reply:
```js
var args = parse('', {
  boolean: ['file'],
  // default: {'file': true},
  configObjects: [{'file': false}]
})
```
before PR #119 the behavior was:
- there is no `--file` in CLI and no user set default, but we set an imlicit false => `file: false`
- set `defaulted[file] = true`
- while processing `configObect` the user input overrides the defaulted value

after PR #119 the behavior is:
- there is no `--file` in CLI and no user set default, so we don't set this property
- therefore we don't set `defaulted` neither
- while processing `configObect` the user input is set
- in case of an user set default: `applyDefaultsAndAliases()` does this job for `number`, `string` and `boolean` the identical way.

IMO this `defaulted` can be explained with historic reasons and can be removed completely.

Reply:
`set-placeholder-key`:
when truthy, `setPlaceholderKeys()` sets the value to `undefinded`, very similar to `applyDefaultsAndAliases()` which sets a user default value instead. So in this case we don't need `defaulted`neither.

Reply:
@juergba do you want to go ahead and submit a patch that completely removes the defaulted behavior then?

Reply:
yes, done.